numba/tests/test_random.py

```diff
@@ -1291,6 +1294,98 @@ def test_multinomial_3_tuple(self):
             self._check_sample(n, pvals, sample)
 
 
+class TestRandomDirichlet(BaseTest):
+    alpha = np.array([1, 1, 1, 2], dtype=np.float64)
+
+    def _check_sample(self, alpha, size, sample):
+        '''Check output properties'''
+        if not size is None:
+
+            if type(size) is tuple:
+                self.assertIsInstance(sample, np.ndarray)
+                self.assertEqual(sample.shape, size + (len(alpha),))
+                self.assertEqual(sample.dtype, np.dtype(np.float64))
+                '''Check statistical properties'''
+                self.assertAlmostEqual(sample.sum(), np.prod(size), places=5)
```

Comment:
This sort of check seems like it might be susceptible to numerical "accidents" e.g.  unlucky RNG generates a pathological sequence that doesn't meet this condition, are there other properties that could perhaps be used instead?

Reply:
Yeah - I've written it so that it now computes the sum along the last axis and makes sure every element of that array is equal to 1.